File: src/string/everyThird/test_every_third.py
from every_third import *
import pytest

data = [
('Python', 'yton'),
('Hello', 'elo'),
('qwer', 'we'),
('p', ''),
('ab', 'b'),
('abc', 'bc'),
('abcdefg', 'bcef'),
('abcdefghij', 'bcefhi'),
('abcdefghijkl', 'bcefhikl'),
('qwertyuiopasdfghjklzxcvbnm', 'wetyioasfgjkzxvbm'),
]


@pytest.mark.parametrize("input, expected", data)
def test_every_third(input, expected):
    assert expected == every_third(input)

@pytest.mark.parametrize("input, expected", data)
def test_every_third_with_del(input, expected):
    assert expected == every_third_with_del(input)

File: src/string/everyThird/TestEveryThird.java
package string.everyThird;

import static org.junit.Assert.assertEquals;

import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.CsvSource;

public class TestEveryThird {
    @ParameterizedTest
    @CsvSource(value = {
            "Python, yton",
            "Hello, elo",
            "qwer, we",
            "p, ''",
            "ab, b",
            "abc, bc",
            "abcdefg, bcef",
            "abcdefghij, bcefhi",
            "abcdefghijkl, bcefhikl",
            "qwertyuiopasdfghjklzxcvbnm, wetyioasfgjkzxvbm",
            "карандаш, арндш",
    })
    void testEveryThird(String input, String expected) {
        assertEquals(expected, EveryThird.everyThird(input));
    }
}
